thunder/dynamo/utils.py

```diff
@@ -418,3 +448,55 @@ def _get_example_inputs_from_placeholder(node) -> tuple[torch.Tensor]:
         raise TypeError(
             "The 'example_value' in the placeholder node is expected to be either a Tensor or a Tuple of Tensors."
         )
+
+
+def _checkpoint_function_converter(gm: torch.fx.GraphModule):
+    """
+    Replace the Torch operators in the GraphModule called by activation checkpoint operator with the corresponding Thunder symbols in place
+    Args:
+        gm: The GraphModule of the checkpointed function, which is modified in place
+    """
```

Comment:
The docstring mentions inplace modification of the GraphModule but there's a deep copy done in the first line of the function.
Why is the graph deepcopied, but not the GraphModule? Should we skip copying the graph and modify it inplace as well?

Reply:
e.g. : the whole split module is as below:
```
# split_module
GraphModule(
  (wrap_body_0): GraphModule() 
  (thunder_1): ThunderModule( 
    (_model): GraphModule()
  )
)
```
the `gm` is `wrap_body_0`, `split_module` has reference to `wrap_body_0`, if we deepcopy `wrap_body_0` we need to update the reference in `split_module`, since we only need to modify the Graph structure, I think update the `gm.graph` inplace and recompile it should be fine

Reply:
Thank you for the explanation. It's totally fine to replace `.graph` with the new object. Can we modify the `.graph` itself inplace then as well? No need to change anything in this PR but we should probably revisit this in a follow-up.